src/content/reducers/profile-view.js

```diff
@@ -347,39 +353,39 @@ export const selectorsForThread = (threadIndex: ThreadIndex): SelectorsForThread
       getCallTreeFilters,
       CallTreeFilters.getCallTreeFilterLabels
     );
-    const getRangeFilteredThread: State => Thread = createSelector(
+    const getRangeFilteredThread = createSelector(
       getThread,
       getDisplayRange,
-      (thread: Thread, range: StartEndRange) => {
+      (thread: Thread, range: StartEndRange): Thread => {
         const { start, end } = range;
         return ProfileData.filterThreadToRange(thread, start, end);
       }
     );
-    const _getRangeFilteredThreadSamples: State => SamplesTable = createSelector(
+    const _getRangeFilteredThreadSamples = createSelector(
       getRangeFilteredThread,
-      thread => thread.samples
+      (thread: Thread): SamplesTable => thread.samples
```

Comment:
I'm fine not typing them if they are inferable. It is pretty easy to follow the functions up for their types.